tools/src/test/scala/scala/scalanative/IncCompilationTest.scala

```diff
@@ -0,0 +1,134 @@
+package scala.scalanative
+
+import org.scalatest.matchers.should.Matchers
+
+import java.io.{File, PrintWriter}
+import java.nio.file.{Files, Path, Paths}
+import scala.scalanative.build.{Config, NativeConfig, _}
+import scala.scalanative.util.Scope
+
+// The test is used for incremental compilation
+
+class IncCompilationTest extends codegen.CodeGenSpec with Matchers {
+  "The test framework" should "generate the llvm IR of object A" in {
+    Scope { implicit in =>
+      val sources = """object A {
+                      |  def print(x: String): Unit = {
+                      |    println(x)
+                      |  }
+                      |  def print(x: Int): Unit = {
+                      |    println(x)
+                      |  }
+                      |  def main(args: Array[String]): Unit = {
+                      |    print(1)
+                      |    print("hello world?")
+                      |  }
+                      |}""".stripMargin
+      val entry = "A"
+      val changedTop = Set[String]("A", "A$")
+      val outDir = Files.createTempDirectory("native-test-out")
+      println(outDir)
+      val compiler = NIRCompiler.getCompiler(outDir)
+      val sourcesDir = NIRCompiler.writeSources(sources)
+      val files = compiler.compile(sourcesDir)
+      makeChanged(outDir, changedTop)
+      val classpath = makeClasspath(outDir)
+      val nativeConfig = build.NativeConfig.empty
+        .withClang(Discover.clang())
+        .withClangPP(Discover.clangpp())
+        .withCompileOptions(Discover.compileOptions())
+        .withLinkingOptions(Discover.linkingOptions())
+        .withLTO(Discover.LTO())
+        .withGC(Discover.GC())
+        .withMode(Discover.mode())
+        .withOptimize(Discover.optimize())
+      val config = makeConfig(outDir, entry, nativeConfig)
+      val ext = if (Platform.isWindows) ".exe" else ""
+      val artifact = outDir.resolve("result" + ext)
+
+      Build.build(config, artifact)
```

Comment:
This part seems to be repeated in different tests. We can factor the common part to a helper method